libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.mapbox.navigation.dropin
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+
+class NavigationViewModel(
+    accessToken: String,
+    application: Application
+) : AndroidViewModel(application) {
```

Comment:
why do we use AndroidViewModel, but not just ViewModel? It adds a bit of magic (you have access to appContext), but it becomes harder to write unit-test.

Reply:
In this case I need context to create an instance of `MapboxNavigation` which is unavailable via the `ViewModel`. We don't want to do that in the view or else it would be destroyed. Do you have any other suggestions to instantiate the `MapboxNavigation` inside `NavigationViewModel` but still use `ViewModel` and get access to context?
cc @RingerJK

Reply:
post-merge comment 😄 
We can make an application provider object, that would be included to `NavigationViewModel` only as an option

Reply:
I have a draft PR where I've done some work on `NavigationViewModel`. I've removed the reference for the Application reference as I don't think it's needed.  It's still subject to review once the PR is ready.